test/alternator/test_streams.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2019 ScyllaDB
+#
+# This file is part of Scylla.
+#
+# Scylla is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Scylla is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+
+# Tests for stream operations: ListStreams, DescribeStream, GetShardIterator.
+
+import pytest
+from botocore.exceptions import ClientError
+from util import list_tables, test_table_name, create_test_table, random_string
+
+stream_types = [ 'OLD_IMAGE', 'NEW_IMAGE', 'KEYS_ONLY', 'NEW_AND_OLD_IMAGES']
+
+def test_list_streams(dynamodb, dynamostreams):
+    for type in stream_types:
+        table = create_test_table(dynamodb,
+            StreamSpecification={'StreamEnabled': True, 'StreamViewType': type},
+            KeySchema=[{ 'AttributeName': 'p', 'KeyType': 'HASH' }],
+            AttributeDefinitions=[{ 'AttributeName': 'p', 'AttributeType': 'S' }]);
+
+        streams = dynamostreams.list_streams(TableName=table.name)
```

Comment:
I only ask for the table generated, and I only care about the first page here (only doing very basic validation). If/when the test becomes more comprehensive, yes, it needs to read the full desc stream.

Reply:
You're right. I didn't notice you only list_streams on the specific table. There indeed there should only be one, so it will never get paged. You're right.
You already wrote the code for implementing paging, so I do think that now, more than later, is a good time to write a test for it. It should be very to test easy as I describe above.